### PR TITLE
4월 DA/ML 블로그 포스팅

### PR DESCRIPTION
**Pull request by Yooonlee**
제목을 잘못 올려서 수정하였습니다.

### CODE REVIEW

File: _posts/2023-04-29-BiT.md [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+tags: ["Bit", "Transfer Learning"]
+image: ../assets/images/post-VAE/img12.png
+featured: true
+use_math: true
+---
+# BiT
+
+안녕하세요. 제가 이번 블로그 게시물에서 소개할 논문은 ECCV 2020에 발표된 Big Transfer(BiT)모델 
+
+입니다.
```
Comment (hjm507) [inline _posts/2023-04-29-BiT.md]: 문장이 끊겨 있어서 붙여줘야 할 것 같습니다.

**Review by hjm507 — CHANGES_REQUESTED** [review]
어렵긴 하지만 전이 학습이 쉽게 접하지는 못하다 보니 흥미로웠습니다. 다만, 후반부부터는 단락 별로 주제가 많이 바뀌기도 하고, 내용이 길다 보니 쉽게 머리에 들어오지 않는 것 같습니다.

 맨 앞 부분에 목차나 배경 지식 등에 관해서 정리해서 미리 언급해주고 이런 순서대로 설명 드리겠다라고 하면 읽기 훨씬 좋을 것 같습니다.

File: _posts/2023-04-29-BiT.md (hunk 1 above)
Comment (Yooonlee) [reply _posts/2023-04-29-BiT.md]: 목차를 만들어 해당사항 수정하였습니다.

File: _posts/2023-04-29-BiT.md [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+[7.Fine-tuning 비교 ](#fine-tuning-비교)
+
+[8.SSL(준지도학습)과 비교](#준지도학습-이란)
+
+
+### Representation Learning
+
+→ <b>Representatioin Learning</b> 이란 <br>feature == semantic feature == representation 를 학습하는 것입니다.
+
+- 본 논문의 메서드는 이전까지 SOTA이상의 높은 정확도를 보입니다.
```
Comment (beautifulchoi) [inline _posts/2023-04-29-BiT.md]: "이전까지의 SOTA 모델 이상의" 로 수정하는게 좋을 것 같습니다

File: _posts/2023-04-29-BiT.md [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+
+## 전이 학습(Transfer Learning)의 개요
+
+- $T_{A}$ = task A
+- $D_A$ = data A
+
+<br>
+<b>학습속도 개선</b>
+
+- 보통 CIFAR-10의 경우 epoch 100~200 이 baseline(기본)인데 5~10번 정도로 해도 충분합니다.
```
Comment (beautifulchoi) [inline _posts/2023-04-29-BiT.md:56]: 이부분 오류인지 제 컴퓨터에서만 그렇게 보이는 건지는 모르겠는데,, 좀 이상하게 표현이 되는 것 같습니다

File: _posts/2023-04-29-BiT.md [hunk 4]
```diff
@@ -0,0 +1,315 @@
...
+    - 가중치 초기화에 대한 민감도를 감소시킵니다.
+    - 모델의 일반화 효과가 있습니다.
+
+![image6](../assets/images/post-BiT/image6.png)
+
+- mini-batch size란 한번에 학습을 위해 들어가는 이미지의 개수입니다.
+- x:  hidden layer에 들어가는 feature , 
+m = hidden layer에 들어가는 feature의 개수입니다.
+- 각 채널(차원)마다 $\gamma$ , $\beta$ 가 다르게 적용됩니다.
+- 감마 : 스케일
```
Comment (beautifulchoi) [inline _posts/2023-04-29-BiT.md:103]: 이 부분도 감마와 베타의 설명이 같은 줄에서 표현이 되고 있어 감마에 대한 설명인 것 같습니다.

Comment (beautifulchoi) [reply _posts/2023-04-29-BiT.md:103]: 감마와 베타를 묶어서 선형결합 파라미터라고 설명하는 것도 좋아보입니다.

File: _posts/2023-04-29-BiT.md [hunk 5]
```diff
@@ -0,0 +1,315 @@
...
+
+ 
+
+### 2. 정규화 기법
+
+- 본 논문에서는 BN을 사용하지 않습니다.
+    - BN 사용 시 모델이 크고, 이미지의 해상도가 높아서
+    장치(gpu device)당 작은 배치를 사용해야 합니다.
+    - 이때, 정확도가 하락하거나 장치간 동기화 비용이 증가하는 문제가 발생합니다.
+- Batch Norm
```
Comment (beautifulchoi) [inline _posts/2023-04-29-BiT.md:137]: 위에서 BN은 Batch normalization 레이어를 넣지 않았다는 것 같은데, 밑에서 설명에 Batch Norm에 대해 설명하고 있어 혼동을 유발할 수 있을 것 같습니다.

File: _posts/2023-04-29-BiT.md [hunk 6]
```diff
@@ -0,0 +1,315 @@
...
+m = hidden layer에 들어가는 feature의 개수입니다.
+- 각 채널(차원)마다 $\gamma$ , $\beta$ 가 다르게 적용됩니다.
+- 감마 : 스케일
+베타 : 데이터 상하좌우 이동 (translate)
+
+![image7](../assets/images/post-BiT/image7.png)
+
+![image8](../assets/images/post-BiT/image8.png)
+
+## Upstream 사전에 좋은 특징 추출기를 만들기 위해 사용
```
Comment (beautifulchoi) [inline _posts/2023-04-29-BiT.md:110]: Upstream 뒤에 : 를 붙이는 것이 메끄러울 것 같습니다

File: _posts/2023-04-29-BiT.md [hunk 7]
```diff
@@ -0,0 +1,315 @@
...
+
+- BiT에서는 사전 학습(pre-training)과 fine-tuning 과정에서 차이점이 존재합니다.
+
+1. Weight-Decay (정규화)
+    
+    가중치에 대해 제약조건을 걸어놓습니다.
+    
+    L2 norm을 많이 사용합니다.
+    
+2. MixUp 2개의 데이터씩 pair로 묶어 2개의 정보를 묶어 사용합니다. (데이터 증진 기법)
```
Comment (beautifulchoi) [inline _posts/2023-04-29-BiT.md:174]: Mixup에 대한 설명은 다음 문장에서 하는 것이 통일감 있을 것 같습니다

File: _posts/2023-04-29-BiT.md [hunk 8]
```diff
@@ -0,0 +1,315 @@
...
+
+- SGD momentum 0.9 , 초기 learning rate = 0.003을 사용합니다.
+pre-training시 정해진 가중치가 바뀔 수 있어 낮은 learning rate를 사용합니다.
+- small task : 500 step만 (500번의 batch만 학습) (epoch아님)
+
+TPUv3-512 : 512개의 chip / 칩당 8개의 이미지가 들어갑니다. 
+
+- 두 과정 모두에서 Learning rate decay를 사용합니다. 
+(학습할수록 learning rate가 줄어드는 방식)
+
```
Comment (beautifulchoi) [inline _posts/2023-04-29-BiT.md:205]: Upstream과 downstream의 finetuning 차이가 간략히 설명되어 있으면 좋을 것 같습니다.

File: _posts/2023-04-29-BiT.md [hunk 9]
```diff
@@ -0,0 +1,315 @@
...
+- 왼쪽
+    - 초록 노랑 : 오히려 BiT가 내린 결정이 타당하다.
+    - 빨강 : BiT가 틀렸다고 판단된다.
+- 오른쪽
+    - Top (위에서 6개) : 모델의 예측이 레이블보다 정확한 경우 (ImageNet클래스의 개수가 1000개다보니 발생함)
+    - Middle (가운데 6개) : Top-5 예측은 맞은 경우
+    - Bottom (아래 6개) :  Top-10 예측이 모두 틀렸으나 레이블이 애매함
+
+![image20](../assets/images/post-BiT/image20.png)
+
```
Comment (beautifulchoi) [inline _posts/2023-04-29-BiT.md:307]: 결론에 대해 간략하게 쓰면 좋을 것 같습니다

**Review by beautifulchoi — COMMENTED** [review]
흥미로운 논문에 대해 글을 작성해주셔서 재미있게 읽었습니다! 글 쓰느라 수고하셨습니다 : )

**Review by hjm507 — DISMISSED** [review]

File: _posts/2023-04-29-BiT.md [hunk 10]
```diff
@@ -0,0 +1,315 @@
...
+[4.[배경지식]  배치 정규화](#[배경지식]-배치-정규화)
+
+[5.Upstream ](#upstream-사전에-좋은-특징-추출기를-만들기-위해-사용)
+
+[6.Downstream ](#downstream-fine-tuning)
+
+[7.Fine-tuning 비교 ](#fine-tuning-비교)
+
+[8.SSL(준지도학습)과 비교](#준지도학습-이란)
```
Comment (hoqn) [inline _posts/2023-04-29-BiT.md]: 목차 걸어주신 거 정말 좋습니다 :)

다만, 의도했던 대로 작동하지 않는 문제가 있습니다.

각 헤딩의 id는 `(`, `)`, `[`, `]`와 같은 특수문자를 **제외**시킵니다! 따라서 링크에서 아래와 같이 이런 문자들을 다 없애시면 의도대로 작동할 것 같습니다.

```suggestion
[1.Representation Learning 소개](#representation-learning)

[2.전이 학습의 개요](#전이-학습transfer-learning의-개요)

[3.[배경지식] 전이 학습 구현 방식](#배경지식-전이-학습-구현-방식)

[4.[배경지식]  배치 정규화](#배경지식-배치-정규화)

[5.Upstream ](#upstream-사전에-좋은-특징-추출기를-만들기-위해-사용)

[6.Downstream ](#downstream-fine-tuning)

[7.Fine-tuning 비교 ](#fine-tuning-비교)

[8.SSL(준지도학습)과 비교](#준지도학습-이란)
```

**Review by hoqn — CHANGES_REQUESTED** [review]

File: _posts/2023-04-29-BiT.md (hunk 4 above)
Comment (Yooonlee) [reply _posts/2023-04-29-BiT.md:103]: 수정하였습니다

File: _posts/2023-04-29-BiT.md (hunk 3 above)
Comment (Yooonlee) [reply _posts/2023-04-29-BiT.md:56]: 이 부분은 저도 이상하게 잘 수정이 안되는 것 같습니다..

File: _posts/2023-04-29-BiT.md (hunk 2 above)
Comment (Yooonlee) [reply _posts/2023-04-29-BiT.md]: 수정하였습니다.

File: _posts/2023-04-29-BiT.md (hunk 5 above)
Comment (Yooonlee) [reply _posts/2023-04-29-BiT.md:137]: 설명 문장을 넣었습니다.

File: _posts/2023-04-29-BiT.md (hunk 8 above)
Comment (Yooonlee) [reply _posts/2023-04-29-BiT.md:205]: 위 189~204에서 설명되었다고 생각했습니다.

File: _posts/2023-04-29-BiT.md (hunk 9 above)
Comment (Yooonlee) [reply _posts/2023-04-29-BiT.md:307]: 수정하였습니다.

File: _posts/2023-04-29-BiT.md (hunk 10 above)
Comment (Yooonlee) [reply _posts/2023-04-29-BiT.md]: 그렇군요.. 몰랐던 사실을 알게해주셔서 감사합니다! 
수정하였습니다.

File: _posts/2023-04-29-BiT.md [hunk 11]
```diff
@@ -0,0 +1,320 @@
...
+<br><br>
+
+## 목차
+[1.Representation Learning 소개](#representation-learning)
+
+[2.전이 학습의 개요](#전이-학습transfer-learning의-개요)
+
+[3.[배경지식] 전이 학습 구현 방식](#배경지식-전이-학습-구현-방식)
+
+[4.[배경지식]  배치 정규화](#배경지식-배치-정규화)
```
Comment (beautifulchoi) [inline _posts/2023-04-29-BiT.md:24]: 여기 부분만 목차에서 링크가 안 걸려 있는 것 같습니다.

Comment (Yooonlee) [reply _posts/2023-04-29-BiT.md:24]: 수정했습니다.

**Review by beautifulchoi — APPROVED** [review]

**Review by hjm507 — APPROVED** [review]
